pkg/adapters/grpc/queue_grpc.go

```diff
@@ -62,7 +65,21 @@ func (s *QueueServiceServer) Send(ctx context.Context, req *pb.QueueSendRequest)
 		Payload:     task.GetPayload().AsMap(),
 	}
 
+	sn := span.FromContext(ctx, "queue-"+req.GetQueue())
+
+	sn.SetAttributes(
+		semconv.CodeFunctionKey.String("Queue.Send"),
+		semconv.MessagingDestinationKindQueue,
+		semconv.MessagingDestinationKey.String(req.GetQueue()),
+		semconv.MessagingMessageIDKey.String(nitricTask.ID),
+		attribute.Key("messaging.message_type").String(nitricTask.PayloadType),
+	)
+
+	defer sn.End()
+
 	if err := s.plugin.Send(req.GetQueue(), nitricTask); err != nil {
+		sn.RecordError(err)
```

Comment:
I'm assuming it's fine for this to run unconditionally?

Reply:
yes, if you haven't set a traceProvider you get a noop trace provider